### ISSUE TITLE
GradientTape problem - Accuracy/Loss differences between fit and eager_tf

### ISSUE THREAD

**Issue by Costyv95** [zzh8829/yolov3-tf2#173]
Hi,

I need to use the eager_tf because I need to apply gradients only on some layers. The problem is I noticed some bad convergence while using eager_tf. 

I then tried to compare  VOC2012 (the example that you gave)  trainings (fit vs eager_tf). It seems that fit converges better without overfitting, while eager_tf overfits and doesn't get to the same losses/accuracy.
I haven't changed your code. Have you managed to make eager_tf get to the same results? I haven't . While struggling to find out what's happening, I found out this:

https://github.com/tensorflow/tensorflow/issues/35585

Hope you know something. Thanks.

**Comment by zzh8829** [comment]
I don't really have to expertise to comment on gradient tape but can you explain a bit more on what apply gradient only on some layer means? maybe you can just tweak some settings in `fit` mode to train that way before the issue gets answered in tensorflow.

**Comment by Costyv95** [comment]
The thing I need to do:

I have a network that looks something like this  A -> B,  A -> C 
A, B, C are some combinations of layers. B, C will output some values

So it's a network with 2 outputs(B, A) and only one input. I have 3 types of updates that I need to do.

1. Get the output from B, and then update with this loss only B weights.
2. Get the output from C, and then update with this loss only C weights.
3. Accumulate the gradients from B and -C and update only the A weights.
 
I could get with freezing for 2 steps, but I don't know exactly how would I would subtract the gradients for step 3.
